app/src/main/java/de/westnordost/streetcomplete/data/osm/osmquests/OsmQuest.kt

```diff
@@ -27,16 +27,23 @@ data class OsmQuest(
     override val type: QuestType<*> get() = osmElementQuestType
 
     override val markerLocations: Collection<LatLon> get() {
-        if (osmElementQuestType.hasMarkersAtEnds && geometry is ElementPolylinesGeometry) {
+        if (geometry is ElementPolylinesGeometry) {
             val polyline = geometry.polylines[0]
             val length = polyline.measuredLength()
-            if (length > 15 * 4) {
-                return listOf(
-                    polyline.pointOnPolylineFromStart(15.0)!!,
-                    polyline.pointOnPolylineFromEnd(15.0)!!
-                )
+            if ((osmElementQuestType.hasMarkersAtEnds && length > MARKER_FROM_END_DISTANCE * 4)
+                || (length > MAXIMUM_MARKER_DISTANCE + 2 * MARKER_FROM_END_DISTANCE)) {
```

Comment:
I am also confused about this algorithm, and I think this is where it's coming from. Which quests don't have markers at the ends, but should have multiple markers along the way? ~~Why do quests with markers along the ends only need 4x the distance from the end before they get the same treatment?~~ I realized this is because if they're sufficiently short, they only get one marker, not on the ends. But I still don't understand why some quests would get markers at both ends even if they're short, and others only start getting that treatment if they're long enough to have markers in the middle, too.

Reply:
Some quests always have markers at their end (speed limits, I think - maybe others) - unless they are very short.

Otherwise, only quests longer than 500 metres will have markers at their end (and then repeated if needed) - so that the mapper is encouraged to map from one end to the other. I found it worked best this way, as only having markers in the middle encouraged me to end up in the middle of the way, and it was harder to map long ways this way without backtracking.

Reply:
Some quests always have markers at their end (speed limits, I think - maybe others) - unless they are very short.

Otherwise, only quests longer than (about) 500 metres will have markers at their end (and then repeated if needed) - so that the mapper is encouraged to map from one end to the other. I found it worked best this way, as only having markers in the middle encouraged me to end up in the middle of the way, and it was harder to map long ways this way without backtracking.

Reply:
I know about the quests with markers at both ends; I was wondering which quests *don't* have them at both ends, but can go longer than 500m. But now I remember it's the surface and cycleway/sidewalk quests, I think.